engines/hypno/arcade.cpp

```diff
@@ -0,0 +1,322 @@
+					}
+				}
+			}
+		}
+
+		uint32 i = 0;
+		shootsToRemove.clear();
+
+		for (Shoots::iterator it = _shoots.begin(); it != _shoots.end(); ++it) {
+			if (it->video->decoder) {
+				int frame = it->video->decoder->getCurFrame();
+				if (frame > 0 && frame >= it->explosionFrame - 15 && !it->destroyed) {
+					hitPlayer();
+				}
+
+				if (frame > 0 && frame >= it->explosionFrame - 3 && !it->destroyed) {
+					skipVideo(*it->video);
+					//_health = _health - it->damage;
+				} else if (frame > 0 && frame >= it->video->decoder->getFrameCount()-2) {
+					skipVideo(*it->video);
+					shootsToRemove.push_back(i);
+				} else if (it->video->decoder->needsUpdate()) {
+					updateScreen(*it->video);
+				}
+			}
+			i++;
+		}
+		if (shootsToRemove.size() > 0) {
+			for (Common::List<uint32>::iterator it = shootsToRemove.begin(); it != shootsToRemove.end(); ++it) {
+				debugC(1, kHypnoDebugArcade, "Removing %d from %d size", *it, _shoots.size());
+				delete _shoots[*it].video;
+				_shoots.remove_at(*it);
+			}
+		}
+
+		if (_music.empty()) {
+			_music = _soundPath + arc.music;
+			playSound(_music, 1);
+		}
+
+		g_system->delayMillis(10);
+	}
+
+	// Deallocate shoots
+	for (Shoots::iterator it = _shoots.begin(); it != _shoots.end(); ++it) {
+		if (it->video->decoder)
+			skipVideo(*it->video);
+		delete it->video;
+	}
+
+	if (background.decoder)
+		skipVideo(background);
+
+}
+
+int HypnoEngine::detectTarget(Common::Point mousePos) {
+	int i = -1;
+	int x = 0;
+	int y = 0;
+	int w = 0;
+	int h = 0;
+	for (Shoots::iterator it = _shoots.begin(); it != _shoots.end(); ++it) {
+		i++;
+		if (it->destroyed || !it->video->decoder)
+			continue;
+		x = mousePos.x - it->position.x;
+		y = mousePos.y - it->position.y;
+		w = it->video->decoder->getWidth();
+		h = it->video->decoder->getHeight();
+		if (it->video->decoder && x >= 0 && y >= 0 && x < w && y < h) {
+			if (it->video->currentFrame->getPixel(x, y) > 0)
+				return i;
+		}
+	}
+	return -1;
+}
+
+void HypnoEngine::drawCursorArcade(Common::Point mousePos) {
+	int i = detectTarget(mousePos);
+	if (i >= 0)
+		changeCursor("target");
+	else
+		changeCursor("arcade");
+		
+	g_system->copyRectToScreen(_compositeSurface->getPixels(), _compositeSurface->pitch, 0, 0, _screenW, _screenH);
+}
+
+bool HypnoEngine::clickedPrimaryShoot(Common::Point mousePos) { return true; };
+
+void HypnoEngine::shoot(Common::Point mousePos) {
+	int i = detectTarget(mousePos);
+	int w = 0;
+	int h = 0;
+	if (i >= 0) {
+		playSound(_soundPath + _shoots[i].endSound, 1);
+		w = _shoots[i].video->decoder->getWidth();
+		h = _shoots[i].video->decoder->getHeight();
+		_score++;
+		_shoots[i].destroyed = true;
+		_shoots[i].video->position = Common::Point(mousePos.x - w / 2, mousePos.y - h / 2);
+		_shoots[i].video->decoder->forceSeekToFrame(_shoots[i].explosionFrame + 2);
+	}
+}
+
+bool HypnoEngine::clickedSecondaryShoot(Common::Point mousePos) {
+	return false;
+}
+
+} // End of namespace Hypno
```

Comment:
Most if not all of your files are missing the last newline

Reply:
Fixed. It's odd that format-clang does not add it if missing, right?